src/parsers/espresso/espresso.js

```diff
@@ -0,0 +1,389 @@
+        element: match[1],
+        mass: parseFloat(match[2]),
+        potential: match[3],
+    }));
+    const atomicPositionsMatches = Array.from(text.matchAll(regex.atomicPositions));
+    const units = text.match(regex.atomicPositionsUnits)[1];
+    const { _units, scalingFactor } = convertFromEspressoUnits(units);
+
+    const elements = atomicPositionsMatches.map((match, index) => ({
+        id: index,
+        value: match[1],
+    }));
+    const coordinates = atomicPositionsMatches.map((match, index) => ({
+        id: index,
+        value: [
+            parseFloat(match[2]) * scalingFactor,
+            parseFloat(match[3]) * scalingFactor,
+            parseFloat(match[4]) * scalingFactor,
+        ],
+    }));
+    const constraints = atomicPositionsMatches
+        .filter((match) => match[5] && match[6] && match[7]) // Check if all three constraints exist
+        .map((match, index) => ({
+            id: index,
+            value: [match[5] === "1", match[6] === "1", match[7] === "1"],
+        }));
+
+    return { elements, coordinates, constraints, units: _units };
+}
+
+/**
+ * @summary Converts ibrav value to cell type according to Quantum Espresso docs
+ * https://www.quantum-espresso.org/Doc/INPUT_PW.html#ibrav
+ * @param {Number} ibrav - ibrav parameter
+ * @returns {String}
+ */
+function ibravToCellType(ibrav) {
+    let type;
+    switch (ibrav) {
+        case 1:
+            type = LATTICE_TYPE.CUB;
+            break;
+        case 2:
+            type = LATTICE_TYPE.FCC;
+            break;
+        case 3:
+            type = LATTICE_TYPE.BCC;
+            break;
+        case -3:
+            type = LATTICE_TYPE.BCC;
+            break;
+        case 4:
+            type = LATTICE_TYPE.HEX;
+            break;
+        case 5:
+            type = LATTICE_TYPE.RHL;
+            break;
+        case -5:
+            type = LATTICE_TYPE.RHL;
+            break;
+        case 6:
+            type = LATTICE_TYPE.TET;
+            break;
+        case 7:
+            type = LATTICE_TYPE.BCT;
+            break;
+        case 8:
+            type = LATTICE_TYPE.ORC;
+            break;
+        case 9:
+            type = LATTICE_TYPE.ORCC;
+            break;
+        case -9:
+            type = LATTICE_TYPE.ORCC;
+            break;
+        case 10:
+            type = LATTICE_TYPE.ORCF;
+            break;
+        case 11:
+            type = LATTICE_TYPE.ORCI;
+            break;
+        case 12:
+            type = LATTICE_TYPE.MCL; // Monoclinic P, unique axis c
+            break;
+        case -12:
+            type = LATTICE_TYPE.MCL; // Monoclinic P, unique axis b
+            break;
+        case 13:
+            type = LATTICE_TYPE.MCLC; // Monoclinic P, base centered
+            break;
+        case -13:
+            type = LATTICE_TYPE.MCLC; // Monoclinic P, unique axis b
+            break;
+        case 14:
+            type = LATTICE_TYPE.TRI;
+            break;
```

Comment:
I'd isolate to an object and put to settings.js here:

```js
IBRAV_TO_LATTICE_TYPE_MAP = {
     "1": LATTICE_TYPE.CUB,
     ...
     "-13": LATTICE_TYPE.MCLC; // Monoclinic P, unique axis b
}
```

then do `return IBRAV_TO_LATTICE_TYPE_MAP[`${ibrav}`]` or throw error

Twice as little code

Reply:
Hm. That's smart. I've adjusted